### PR TITLE
Rename VirtualMachine to VirtualMachineInstance and OfflineVirtualMachine to VirtualMachine

### PR DESCRIPTION
**Pull request by rmohr**
- [x] Rename VirtualMachine to VirtualMachineInstance
 - [x] Ensure that all examples use VirtualMachineInstance where appropriate
 - [x] Rename vm variables to vmi
 - [x] Rename OfflineVirtualMachine to VirtualMachine
 - [x] Ensure that all examples use VirtualMachine where appropriate
 - [x] Rename ovm variables to vm
 - [ ] Go through doc strings and generated apis and check for inconsistent usage
 - [x] Bump API version to `v1alpha2`

Side-effects:

 * `VirtualMachineReplicaSets` are now called `VirtualMachineInstanceReplicaSets`
 * `VirtualMachinePresets` are now called `VirtualMachineInstancePresets`

Another take of #1020, implementing the result of this discusson: https://groups.google.com/forum/#!topic/kubevirt-dev/CU_VskPIisg

### CODE REVIEW

**Comment by rmohr** [comment]
Still have to go through docs and comments but new structure should now be in the final shape. @davidvossel @cynepco3hahue @vladikr please review.

File: pkg/kubecli/ovm.go [hunk 1]
```diff
@@ -27,11 +27,11 @@ import (
 	"kubevirt.io/kubevirt/pkg/api/v1"
 )
 
-func (k *kubevirt) OfflineVirtualMachine(namespace string) OfflineVirtualMachineInterface {
+func (k *kubevirt) VirtualMachine(namespace string) VirtualMachineInterface {
 	return &ovm{
```
Comment (davidvossel) [inline pkg/kubecli/ovm.go]: ovm

File: pkg/kubecli/ovm.go [hunk 2]
```diff
@@ -41,24 +41,24 @@ type ovm struct {
 	resource   string
 }
 
-// Create new OfflineVirtualMachine in the cluster to specified namespace
-func (o *ovm) Create(offlinevm *v1.OfflineVirtualMachine) (*v1.OfflineVirtualMachine, error) {
-	newOvm := &v1.OfflineVirtualMachine{}
+// Create new VirtualMachine in the cluster to specified namespace
+func (o *ovm) Create(offlinevm *v1.VirtualMachine) (*v1.VirtualMachine, error) {
```
Comment (davidvossel) [inline pkg/kubecli/ovm.go]: ovm

File: pkg/kubecli/ovm.go [hunk 3]
```diff
@@ -41,24 +41,24 @@ type ovm struct {
...
+	newOvm := &v1.VirtualMachine{}
```
Comment (davidvossel) [inline pkg/kubecli/ovm.go]: ovm

File: pkg/kubecli/ovm.go [hunk 4]
```diff
@@ -41,24 +41,24 @@ type ovm struct {
...
 	err := o.restClient.Post().
 		Resource(o.resource).
 		Namespace(o.namespace).
 		Body(offlinevm).
 		Do().
 		Into(newOvm)
```
Comment (davidvossel) [inline pkg/kubecli/ovm.go]: ovm

File: pkg/kubecli/ovm.go [hunk 5]
```diff
@@ -41,24 +41,24 @@ type ovm struct {
...
 
-	newOvm.SetGroupVersionKind(v1.OfflineVirtualMachineGroupVersionKind)
+	newOvm.SetGroupVersionKind(v1.VirtualMachineGroupVersionKind)
```
Comment (davidvossel) [inline pkg/kubecli/ovm.go]: lots more ovm, run a filter over this file

File: pkg/kubecli/ovm_test.go [hunk 6]
```diff
@@ -49,77 +49,77 @@ var _ = Describe("Kubevirt OfflineVirtualMachine Client", func() {
 		Expect(err).ToNot(HaveOccurred())
 	})
 
-	It("should fetch a VM", func() {
+	It("should fetch a VirtualMachineInstance", func() {
 		ovm := NewMinimalOVM("testvm")
```
Comment (davidvossel) [inline pkg/kubecli/ovm_test.go]: NewMinimalOVM test

run ovm to vm filter over this file

File: pkg/kubecli/ovm_test.go [hunk 7]
```diff
@@ -130,10 +130,10 @@ var _ = Describe("Kubevirt OfflineVirtualMachine Client", func() {
...
 
-func NewMinimalOVM(name string) *v1.OfflineVirtualMachine {
-	return &v1.OfflineVirtualMachine{TypeMeta: k8smetav1.TypeMeta{APIVersion: v1.GroupVersion.String(), Kind: "OfflineVirtualMachine"}, ObjectMeta: k8smetav1.ObjectMeta{Name: name}}
+func NewMinimalOVM(name string) *v1.VirtualMachine {
+	return &v1.VirtualMachine{TypeMeta: k8smetav1.TypeMeta{APIVersion: v1.GroupVersion.String(), Kind: "VirtualMachine"}, ObjectMeta: k8smetav1.ObjectMeta{Name: name}}
 }
 
-func NewOVMList(ovms ...v1.OfflineVirtualMachine) *v1.OfflineVirtualMachineList {
-	return &v1.OfflineVirtualMachineList{TypeMeta: k8smetav1.TypeMeta{APIVersion: v1.GroupVersion.String(), Kind: "OfflineVirtualMachineList"}, Items: ovms}
+func NewOVMList(ovms ...v1.VirtualMachine) *v1.VirtualMachineList {
```
Comment (davidvossel) [inline pkg/kubecli/ovm_test.go]: NewOVMList

File: pkg/virt-api/api.go [hunk 8]
```diff
@@ -141,32 +141,32 @@ func (app *virtAPIApp) Execute() {
...
 func (app *virtAPIApp) composeResources(ctx context.Context) {
 
-	vmGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachines"}
-	vmrsGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachinereplicasets"}
-	vmpGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachinepresets"}
-	ovmGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "offlinevirtualmachines"}
+	vmGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachineinstances"}
+	vmrsGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachineinstancereplicasets"}
+	vmpGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachineinstancepresets"}
+	ovmGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachines"}
```
Comment (davidvossel) [inline pkg/virt-api/api.go]: ovm

File: pkg/virt-api/api.go [hunk 9]
```diff
@@ -141,32 +141,32 @@ func (app *virtAPIApp) Execute() {
 
 func (app *virtAPIApp) composeResources(ctx context.Context) {
 
-	vmGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachines"}
-	vmrsGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachinereplicasets"}
-	vmpGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachinepresets"}
-	ovmGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "offlinevirtualmachines"}
+	vmGVR := schema.GroupVersionResource{Group: v1.GroupVersion.Group, Version: v1.GroupVersion.Version, Resource: "virtualmachineinstances"}
```
Comment (davidvossel) [inline pkg/virt-api/api.go]: vm vs vmi

File: pkg/virt-api/api.go [hunk 10]
```diff
@@ -511,8 +511,8 @@ func (app *virtAPIApp) getSelfSignedCert() error {
 func (app *virtAPIApp) createWebhook() error {
 	namespace := getNamespace()
 	registerWebhook := false
-	vmPath := vmValidatePath
-	ovmPath := ovmValidatePath
+	vmPath := vmiValidatePath
+	ovmPath := vmValidatePath
```
Comment (davidvossel) [inline pkg/virt-api/api.go]: ovm

File: pkg/virt-api/api.go [hunk 11]
```diff
@@ -634,10 +634,10 @@ func (app *virtAPIApp) createWebhook() error {
 		}
 	}
 
-	http.HandleFunc(vmValidatePath, func(w http.ResponseWriter, r *http.Request) {
+	http.HandleFunc(vmiValidatePath, func(w http.ResponseWriter, r *http.Request) {
 		validating_webhook.ServeVMs(w, r)
 	})
-	http.HandleFunc(ovmValidatePath, func(w http.ResponseWriter, r *http.Request) {
+	http.HandleFunc(vmValidatePath, func(w http.ResponseWriter, r *http.Request) {
 		validating_webhook.ServeOVMs(w, r)
```
Comment (davidvossel) [inline pkg/virt-api/api.go]: ServeOVMs

File: pkg/virt-api/validating-webhook/validating-webhook.go [hunk 12]
```diff
@@ -461,24 +461,24 @@ func ServeVMs(resp http.ResponseWriter, req *http.Request) {
 
 func admitOVMs(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
```
Comment (davidvossel) [inline pkg/virt-api/validating-webhook/validating-webhook.go]: admitOVMs

File: pkg/virt-api/validating-webhook/validating-webhook_test.go [hunk 13]
```diff
@@ -178,14 +178,14 @@ var _ = Describe("Validating Webhook", func() {
 	})
 
 	Context("with OVM admission review", func() {
```
Comment (davidvossel) [inline pkg/virt-api/validating-webhook/validating-webhook_test.go]: run ovm filter over this file

File: tests/ovm_test.go [hunk 14]
```diff
@@ -89,13 +89,13 @@ var _ = Describe("OfflineVirtualMachine", func() {
 				Name:       "testdisk",
 				VolumeName: "testvolume",
 			})
-			newOVM := NewRandomOfflineVirtualMachine(template, false)
+			newOVM := NewRandomVirtualMachine(template, false)
```
Comment (davidvossel) [inline tests/ovm_test.go]: ovm. run ovm filter here

File: tools/crd-generator/crd-generator.go [hunk 15]
```diff
@@ -69,19 +69,19 @@ func generateOfflineVirtualMachineCrd() {
...
 
 		Names: extensionsv1.CustomResourceDefinitionNames{
-			Plural:     "virtualmachinepresets",
-			Singular:   "virtualmachinepreset",
-			Kind:       v1.VirtualMachinePresetGroupVersionKind.Kind,
-			ShortNames: []string{"vmpreset", "vmpresets"},
+			Plural:     "virtualmachineinstancepresets",
+			Singular:   "virtualmachineinstancepreset",
+			Kind:       v1.VirtualMachineInstancePresetGroupVersionKind.Kind,
+			ShortNames: []string{"vmipreset", "vmipresets"},
```
Comment (davidvossel) [inline tools/crd-generator/crd-generator.go]: another discussion point.  we might need presets to interact with VMs instead of VMIs

**Comment by rmohr** [comment]
@davidvossel comments should be addressed and variables should be renamed now, need another rebase because of todays merges, but apart from that worth another review.

File: docs/devel/offline-virtual-machine.md [hunk 16]
```diff
@@ -1,71 +1,71 @@
-# Offline Virtual Machine developer documentation
+# Virtual Machine developer documentation
```
Comment (cynepco3hahue) [inline docs/devel/offline-virtual-machine.md]: We need also rename the file

File: docs/replica-sets.md [hunk 17]
```diff
@@ -1,27 +1,27 @@
-# VirtualMachineReplicaSet
+# VirtualMachineInstanceReplicaSet
```
Comment (cynepco3hahue) [inline docs/replica-sets.md]: The same rename file

File: pkg/kubecli/ovm.go [hunk 18]
```diff
@@ -27,69 +27,69 @@ import (
 	"kubevirt.io/kubevirt/pkg/api/v1"
```
Comment (cynepco3hahue) [inline pkg/kubecli/ovm.go]: We need to rename file

File: pkg/kubecli/ovm_test.go [hunk 19]
```diff
@@ -35,12 +35,12 @@ import (
 	"kubevirt.io/kubevirt/pkg/api/v1"
```
Comment (cynepco3hahue) [inline pkg/kubecli/ovm_test.go]: We need to rename file

**Comment by rmohr** [comment]
Code is rebased and files are now renamed too. @cynepco3hahue @davidvossel @davidvossel please do another review.

**Comment by davidvossel** [comment]
retest this please

**Comment by rmohr** [comment]
@davidvossel No should be good.

**Pull request merged**